[occm] Runtime panic when creating neutron-based LB service

/kind bug

**What happened**:
`openstack-cloud-controller-manager` pod crashes whenever a LoadBalancer service is created with the following error stack trace :
```
I0818 13:17:15.455652       1 event.go:278] Event(v1.ObjectReference{Kind:"Service", Namespace:"test", Name:"nginxservice-lb", UID:"d186c8c0-8ac9-40d2-9a18-6de8f42aaa3e", APIVersion:"v1", ResourceVersion:"1292391", FieldPath:""}): type: 'Normal' reason: 'EnsuringLoadBalancer' Ensuring load balancer
E0818 13:17:16.095636       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 293 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1934700, 0x2ca6a70)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1934700, 0x2ca6a70)
	/opt/go/src/runtime/panic.go:967 +0x15d
k8s.io/cloud-provider-openstack/pkg/cloudprovider/providers/openstack.(*LbaasV2).createLoadBalancer(0xc00077c8c0, 0xc000130350, 0xc0005d5080, 0x2c, 0x1ba9502, 0xa, 0x0, 0xc0003a0701, 0x0, 0x0, ...)
	/home/ubuntu/go/src/k8s.io/cloud-provider-openstack/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:481 +0x361
k8s.io/cloud-provider-openstack/pkg/cloudprovider/providers/openstack.(*LbaasV2).EnsureLoadBalancer(0xc00077c8c0, 0x1ea8e00, 0xc000058028, 0x1ba9502, 0xa, 0xc000130350, 0xc000693ca0, 0x3, 0x4, 0x1ba38b8, ...)
	/home/ubuntu/go/src/k8s.io/cloud-provider-openstack/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:942 +0xa4d
k8s.io/kubernetes/pkg/controller/service.(*Controller).ensureLoadBalancer(0xc00077c700, 0xc000130350, 0x0, 0x0, 0x6)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:389 +0xde
k8s.io/kubernetes/pkg/controller/service.(*Controller).syncLoadBalancerIfNeeded(0xc00077c700, 0xc000130350, 0xc00020d7a0, 0x14, 0xc000693c40, 0xc000976ca0, 0x126d973)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:344 +0x879
k8s.io/kubernetes/pkg/controller/service.(*Controller).processServiceCreateOrUpdate(0xc00077c700, 0xc000130350, 0xc00020d7a0, 0x14, 0x0, 0x0)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:280 +0xdf
k8s.io/kubernetes/pkg/controller/service.(*Controller).syncService(0xc00077c700, 0xc00020d7a0, 0x14, 0x0, 0x0)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:759 +0x30a
k8s.io/kubernetes/pkg/controller/service.(*Controller).processNextWorkItem(0xc00077c700, 0x203000)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:238 +0xf0
k8s.io/kubernetes/pkg/controller/service.(*Controller).worker(0xc00077c700)
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:227 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000422860)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000422860, 0x1e60600, 0xc000696f00, 0xc00051a101, 0xc0006fc000)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000422860, 0x3b9aca00, 0x0, 0x17734652f485801, 0xc0006fc000)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/wait/wait.go:133 +0xe2
k8s.io/apimachinery/pkg/util/wait.Until(0xc000422860, 0x3b9aca00, 0xc0006fc000)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.18.0/pkg/util/wait/wait.go:90 +0x4d
created by k8s.io/kubernetes/pkg/controller/service.(*Controller).Run
	/home/ubuntu/go/pkg/mod/k8s.io/kubernetes@v1.18.0/pkg/controller/service/controller.go:216 +0x205
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x138cb91]
```
service remains in pending state :
```
NAME              TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
nginxservice-lb   LoadBalancer   10.233.54.249   <pending>     80:32477/TCP   2m30s
```

**What you expected to happen**:
I would expect the occm pod not to crash, the service to be created with an external IP in the subnet provided.

**How to reproduce it**:
Openstack cloud environment is hosted and managed by our provider.
K8s cluster is deployed using kubespray and external cloud provider for openstack.
K8s workers have 2 network interfaces : 1 internal and 1 shared (routed accross several cloud environments)
K8s nodes' internal port have additional allowed-addresses for 10.233.0.0/18 and 10.233.64.0/18.

LBaaS is configured with :
```
[LoadBalancer]
use-octavia=false
create-monitor=False
monitor-delay=1m
monitor-timeout=30s
monitor-max-retries=3
network-id=<id of the shared network>
subnet-id=<id of the shared subnet>
```

Here is the manifest for the nginx deployment+service :
```
apiVersion: v1
kind: Namespace
metadata:
  name: test
---
apiVersion: v1
kind: Pod
metadata:
  name: nginx
  namespace: test
  labels:
   app: nginx
spec:
  containers:
  - name: nginx
    image: nginx
    ports:
    - containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: nginx
  name: nginxservice-lb
  namespace: test
spec:
  ports:
  - port: 80
    targetPort: 80
    protocol: TCP
  selector:
    app: nginx
  type: LoadBalancer
```

**Anything else we need to know?**:
Using annotations in the service manifest doesn't help and occm pod crashes the exact same way when configuring the loadbalancer in the internal network/subnet.
Both shared and internal networks have only one subnet.

**Environment**:
- openstack-cloud-controller-manager version: v1.18.1 (docker.io/k8scloudprovider/openstack-cloud-controller-manager:v1.18.1)
- OpenStack version: v18 Rocky with Neutron LBaaS v2 extension
- Others:
  - kubespray [@5a31123](https://github.com/kubernetes-sigs/kubespray/commit/5a311236c402dbe0a2b8ed9fbfb5f2eb5472b77b)

--- a/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go
+++ b/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go
@@ -475,15 +475,15 @@ func (lbaas *LbaasV2) createLoadBalancer(service *corev1.Service, name, clusterN
 	}
 
 	loadbalancer, err := loadbalancers.Create(lbaas.lb, createOpts).Extract()
+	if err != nil {
+		return nil, fmt.Errorf("error creating loadbalancer %v: %v", createOpts, err)
+	}
 
 	// when NetworkID is specified, subnet will be selected by the backend for allocating virtual IP
 	if (lbClass != nil && lbClass.NetworkID != "") || lbaas.opts.NetworkID != "" {
 		lbaas.opts.SubnetID = loadbalancer.VipSubnetID
 	}
 
-	if err != nil {
-		return nil, fmt.Errorf("error creating loadbalancer %v: %v", createOpts, err)
-	}
 	return loadbalancer, nil
 }
 

